xtc/src/cycles_wallet.rs

```diff
@@ -46,6 +47,7 @@ async fn call(args: CallCanisterArgs) -> Result<CallResult, String> {
         return Err("Attempted to call forward on self. This is not allowed.".to_string());
     }
 
+    let deduced_fee = compute_fee(args.cycles);
     let ledger = ic.get_mut::<Ledger>();
     ledger
         .withdraw(&caller, args.cycles)
```

Comment:
shouldn't we deduct the fee here? There is no guarantee that after .call_raw there will be enough money on the account

Reply:
yeah this was an error that I also noticed when adding unit test. Thank you!